src/vm/exceptionhandling.cpp

```diff
@@ -5031,65 +5032,67 @@ bool IsDivByZeroAnIntegerOverflow(PCONTEXT pContext)
 }
 #endif //_AMD64_
 
+BOOL PALAPI IsSafeToHandleHardwareException(PCONTEXT contextRecord, PEXCEPTION_RECORD exceptionRecord)
+{
+    PCODE controlPc = GetIP(contextRecord);
+    return g_fEEStarted && (
+        exceptionRecord->ExceptionCode == STATUS_BREAKPOINT || 
+        exceptionRecord->ExceptionCode == STATUS_SINGLE_STEP ||
+        ExecutionManager::IsManagedCode(controlPc) || 
```

Comment:
`ExecutionManager::IsManagedCode` will take a lock unless we are in cooperative mode already. Do we need to check whether we are in cooperative mode before calling it?